app/components-react/widgets/CustomWidget.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { IWidgetCommonState, WidgetModule } from './common/useWidget';
+import { WidgetLayout } from './common/WidgetLayout';
+
+interface ICustomWidgetState extends IWidgetCommonState {
+  data: {
+    settings: {
+      custom_enabled: false;
+      custom_html: '';
+      custom_css: '';
+      custom_js: '';
+      custom_json: null;
+    };
+  };
+}
+
+export function CustomWidget() {
+  return (
+    <WidgetLayout>
+      <></>
+    </WidgetLayout>
+  );
+}
+
+export class CustomWidgetModule extends WidgetModule<ICustomWidgetState> {
+  patchAfterFetch(data: any): ICustomWidgetState {
+    return {
+      ...data,
+      settings: {
+        ...data.settings,
+        custom_enabled: data.settings.custom_enabled,
+        custom_html: data.settings.custom_html,
+        custom_css: data.settings.custom_css,
+        custom_js: data.settings.custom_js,
+        custom_json: data.settings.custom_json,
+      },
+    };
+  }
+
+  patchBeforeSend(settings: ICustomWidgetState['data']['settings']): any {
+    return {
+      ...settings,
+      custom_enabled: settings.custom_enabled,
+      custom_html: settings.custom_html,
+      custom_css: settings.custom_css,
+      custom_js: settings.custom_js,
+      custom_json: settings.custom_json,
+    };
+  }
```

Comment:
I don't think we need these `patchAfter` and `patchBefore` functions here:

We use these functions when the api sends us data in a format that's un-standardized or less useful to transform the data into a "nicer" shape for desktop. Notably things like converting ms to s, or making nested data flatter. Here it doesn't look like we're preforming any transforming, so we can just omit these functions :)

Reply:
Sounds good - thanks!